packages/react-components/source/react/library/select/SelectTarget.js

```diff
@@ -18,23 +19,26 @@ const renderText = (type, value, placeholder) => {
 const SelectTarget = forwardRef(
   ({ error, value, type, placeholder, className, ...rest }, ref) => (
     <div className={classNames('rc-input-container', 'rc-select-target')}>
-      <button
-        type="button"
+      <Button
+        type={type === 'textselect' ? 'text' : null}
         className={classNames('rc-input', {
           'rc-input-error': error,
           'rc-input-empty': !value,
+          'rc-input-text-select': type === 'textselect',
         })}
         ref={ref}
         {...rest}
       >
         <Icon
-          className="rc-input-icon trailing"
+          className={classNames('rc-input-icon trailing', {
+            'rc-input-icon-text-select': type === 'textselect',
+          })}
           width="16px"
           height="16px"
           type="chevron-down"
         />
         {renderText(type, value, placeholder)}
-      </button>
+      </Button>
```

Comment:
Here we do 3 separate boolean checks for `type === textselect`. I think we should just create two different elements based on `type === textselect` and pass it into the jsx:

```
let divContents = null;
if (type === 'textelect') {
    divContents = <>yourNewTextSelectStuff</>
} else {
    divContents = <>existingdivContenst</>
}

return <div>{divContents}</div>
```

This way its _really_ hard for a future dev to mess up the boolean logic when modifying the code.